src/Corrade/Utility/Directory.cpp

```diff
@@ -202,23 +202,39 @@ bool mkpath(const std::string& path) {
     if(path.back() == '/')
         return mkpath(path.substr(0, path.size()-1));
 
-    /* If the directory exists, done */
-    if(exists(path)) return true;
-
     /* If parent directory doesn't exist, create it */
     const std::string parentPath = Directory::path(path);
-    if(!parentPath.empty() && !exists(parentPath) && !mkpath(parentPath)) return false;
+    if(!parentPath.empty() && !exists(parentPath) && !mkpath(parentPath)) {
+        Error{} << "Utility::Directory::mkpath(): error: could not create parent path";
+        return false;
+    }
 
     /* Create directory, return true if successfully created or already exists */
 
     /* Unix, Emscripten */
     #if defined(CORRADE_TARGET_UNIX) || defined(CORRADE_TARGET_EMSCRIPTEN)
     const int ret = mkdir(path.data(), 0777);
-    return ret == 0;
+    if(ret != 0 && errno != EEXIST) {
+        Error{} << "Utility::Directory::mkpath(): error:" << strerror(errno);
+        return false;
+    }
+    return true;
 
     /* Windows (not Store/Phone) */
     #elif defined(CORRADE_TARGET_WINDOWS) && !defined(CORRADE_TARGET_WINDOWS_RT)
-    return CreateDirectoryW(widen(path).data(), nullptr) != 0 || GetLastError() == ERROR_ALREADY_EXISTS;
+    if(CreateDirectoryW(widen(path).data(), nullptr) == 0 && GetLastError() != ERROR_ALREADY_EXISTS) {
+        WCHAR* errorStringW = nullptr;
+        FormatMessage(FORMAT_MESSAGE_FROM_SYSTEM|FORMAT_MESSAGE_ALLOCATE_BUFFER,
+            nullptr, GetLastError(), 0, reinterpret_cast<LPWSTR>(&errorStringW),
+            0, nullptr);
+        Containers::ScopeGuard e{errorStringW, LocalFree};
+        std::string errorStringUtf8 = Unicode::narrow(errorStringW);
+        /* The error message already contains a newline */
```

Comment:
There's a flag to disable the newline, I'd rather see that than disabling the newline on our side :)

Reply:
Are you sure? I found a format escape sequence, but that doesn't work with `FORMAT_MESSAGE_FROM_SYSTEM`, since it doesn't use the format string 🤔

Reply:
Eh FFS, what an API :roll_eyes: Sorry, missed that. 

Alternative solution -- cut the last character off :)

Reply:
Sure 👍